validation.go

```diff
@@ -149,3 +190,71 @@ func convertValidations(validations []Validation) []cpluginv1.ParameterValidatio
 	}
 	return out
 }
+
+// ApplyConfigValidations a utility function that applies all the validations for parameters, return an error that
+// consists of a combination of errors from the configurations.
+func applyConfigValidations(params map[string]Parameter, config map[string]string) error {
+	multiErr := assignParamDefaults(params, config)
+
+	for pKey, param := range params {
+		if config[pKey] != "" {
+			multiErr = multierr.Append(multiErr, validateParamType(param, pKey, config[pKey]))
+		}
+		multiErr = multierr.Append(multiErr, validateParamValue(pKey, config[pKey], param))
+	}
+	return multiErr
+}
+
+// validateParamValue validates that a configuration value matches all the validations required for the parameter.
+func validateParamValue(key string, value string, param Parameter) error {
+	var multiErr error
+	var err error
+
+	for _, v := range param.Validations {
+		err = v.validate(value)
+		if err != nil {
+			multiErr = multierr.Append(multiErr, fmt.Errorf("error validating %q: %w", key, err))
+		}
+	}
+	return multiErr
+}
+
+// assignParamDefaults fills any empty configuration with its assigned default value
+// returns an error if a parameter is not recognized.
+func assignParamDefaults(params map[string]Parameter, config map[string]string) error {
```

Comment:
WDYT about not mutating the input configuration map and instead returning the supplemented map?

Reply:
either way is fine... is it a better convention or a preference?

Reply:
IMHO, mutating parameters can lead to surprise for the caller of a method, but I also acknowledge that it's convenient. At least I'm leaning towards not mutating input parameters, but I wouldn't say that's a convention that should be followed.

Reply:
done